examples/seismic/source.py

```diff
@@ -8,53 +10,100 @@
 except:
     plt = None
 
-__all__ = ['PointSource', 'Receiver', 'Shot', 'RickerSource', 'GaborSource']
+__all__ = ['PointSource', 'Receiver', 'Shot', 'WaveletSource',
+           'RickerSource', 'GaborSource']
 
 
 class PointSource(SparseTimeFunction):
     """Symbolic data object for a set of sparse point sources
 
-    :param name: Name of the symbol representing this source
+    :param name: Name of the symbol representing this source.
     :param grid: :class:`Grid` object defining the computational domain.
-    :param coordinates: Point coordinates for this source
-    :param data: (Optional) Data values to initialise point data
-    :param ntime: (Optional) Number of timesteps for which to allocate data
-    :param npoint: (Optional) Number of sparse points represented by this source
-    :param time_order: (Optional) Time discretization order (defaults to 2)
+    :param coordinates: Point coordinates for this source.
+    :param time_range: (Optional) Tuple of time origin, time interval and
+                       number of timesteps, (t0, dt, ntime). Can only set
+                       one of time_range and time.
+    :param time: (Optional) Time series. Can only set one of time_range and time.
+    :param data: (Optional) Data values to initialise point data.
+    :param npoint: (Optional) Number of sparse points represented by this source.
+    :param time_order: (Optional) Time discretization order (defaults to 2).
     :param dimension: :(Optional) class:`Dimension` object for
-                       representing the number of points in this source
+                       representing the number of points in this source.
 
     Note, either the dimensions `ntime` and `npoint` or the fully
     initialised `data` array need to be provided.
     """
 
-    def __new__(cls, name, grid, ntime=None, npoint=None, data=None,
-                coordinates=None, **kwargs):
+    def __new__(self, name, grid, time_range=None, time=None, npoint=None,
```

Comment:
the convention is that ``self`` is used as a reference to the object; ``cls`` is the class type. ``new`` should return an instance of ``cls``. So here you should probably stick to ``cls``, not ``self``

Reply:
Done